platform/settings.cpp

```diff
@@ -17,375 +17,407 @@
 #include "std/iostream.hpp"
 #include "std/sstream.hpp"
 
-
 #define FIRST_LAUNCH_KEY "FirstLaunchOnDate"
 
-static char const DELIM_CHAR = '=';
+namespace
+{
+char const DELIM_CHAR = '=';
+}
+
+char const * Settings::kLocationStateMode = "LastLocationStateMode";
 
 namespace Settings
 {
-  StringStorage::StringStorage()
+StringStorage::StringStorage()
+{
+  lock_guard<mutex> guard(m_mutex);
+
```

Comment:
Do we really need to take a lock in the ctor? Is it possible to use this class before it has been constructed?